crates/holochain/src/conductor/conductor.rs

```diff
@@ -663,10 +685,44 @@ where
 
         let p2p_kv = AgentKv::new(environ.clone().into())?;
         let env = environ.guard();
-        let reader = env.reader()?;
+        let mut out = Vec::new();
+        let mut expired = Vec::new();
+
+        {
+            let reader = env.reader()?;
+
+            let mut iter = p2p_kv.as_store_ref().iter(&reader)?;
+
+            let now: u64 = std::time::SystemTime::now()
+                .duration_since(std::time::UNIX_EPOCH)
+                .unwrap()
+                .as_millis() as u64;
+
+            loop {
+                match iter.next() {
+                    Ok(Some((k, v))) => {
+                        let info = kitsune_p2p::agent_store::AgentInfo::try_from(&v)?;
+                        if info.signed_at_ms() + info.expires_after_ms() <= now {
+                            expired.push(AgentKvKey::from(k));
+                        } else {
+                            out.push(v);
+                        }
+                    }
+                    Ok(None) => break,
+                    Err(e) => return Err(e.into()),
+                }
+            }
+        }
+
+        if !expired.is_empty() {
+            env.with_commit(|writer| {
```

Comment:
I think this has the same issue as above.

Reply:
ditto